pkg/logs/input/docker/tailer.go

```diff
@@ -106,12 +109,17 @@ func (t *Tailer) setupReader(since string) (io.ReadCloser, context.CancelFunc, e
 	}
 	ctx, cancelFunc := context.WithCancel(context.Background())
 	reader, err := t.cli.ContainerLogs(ctx, t.ContainerID, options)
-	return reader, cancelFunc, err
+	t.mutex.Lock()
```

Comment:
can we call `setupReader` concurrently ?

Reply:
it's only concurrent with the forwarder in terms of `lastSince` access
and might be between `Stop` and `setupReader` on the reader access 🤔